rasa/core/nlg/response.py

```diff
@@ -25,28 +25,54 @@ def __init__(self, responses: Dict[Text, List[Dict[Text, Any]]]) -> None:
         """
         self.responses = responses
 
+    def _matches_filled_slots(
+        self, filled_slots: Dict[Text, Any], response: Dict[Text, Any],
+    ) -> bool:
+        """Checks if the conditional response variation matches the filled slots."""
+        constraints = response.get("condition")
+        for constraint in constraints:
+            name = constraint["name"]
+            value = constraint["value"]
+            if filled_slots.get(name) != value:
+                return False
+
+        return True
+
     def _responses_for_utter_action(
-        self, utter_action: Text, output_channel: Text
+        self, utter_action: Text, output_channel: Text, filled_slots: Dict[Text, Any]
     ) -> List[Dict[Text, Any]]:
-        """Return array of responses that fit the channel and action."""
-        channel_responses = []
+        """Returns array of responses that fit the channel, action and condition."""
         default_responses = []
+        conditional_responses = []
 
         for response in self.responses[utter_action]:
-            if response.get("channel") == output_channel:
-                channel_responses.append(response)
-            elif not response.get("channel"):
+            if response.get("condition") is None:
```

Comment:
optional: might be worth making `condition` a constant. 2 usages is kind of at the edge but I think it's likely that Rasa X might need it as well.

Reply:
Yup, can definitely action this - is there a certain file I should turn this into a constant? Or at the top of `response.py` is fine as well?

Reply:
I'd go for `rasa.shared.constants` or `rasa.shared.nlu.constants`. From the point of this PR it's fine to define it at the top of this file, but considering Rasa X and also that you'll need that for sanity checks in the domain sooner or later, I'd go for a `constants` file

Reply:
Cool, I'll define it in `rasa.shared.constants` 👍